### PR TITLE
chore: Add support for TLS tests to the contract test service.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: test-service/rebar.config [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {erl_opts, [debug_info]}.
 {deps, [
-    {shotgun, "0.5.0"},
```
Comment (kinyoklion) [inline test-service/rebar.config:3]: Needed to use 1.0.1 to support TLS as implementeed in the SDK. This should have been updated earlier.

File: test-service/rebar.config [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+    {shotgun, "1.0.1"},
     {jsx, "3.1.0"},
     {verl, "1.0.1"},
     {lru, "2.4.0"},
     {backoff, "1.1.6"},
     {uuid, "2.0.2", {pkg, uuid_erl}},
-    {eredis, "1.4.0"},
-    {yamerl, "0.8.1"},
-    {certifi, "2.8.0"},
+    {eredis, "1.7.1"},
```
Comment (kinyoklion) [inline test-service/rebar.config:9]: Updating other dependencies to match the SDK.

File: test-service/src/ts_sdk_config_params.erl [hunk 3]
```diff
@@ -234,3 +254,49 @@ add_tag_application_version(_, Options) -> Options.
 
 application_or_empty(#{application := Application} = _Options) -> Application;
 application_or_empty(_Options) -> #{}.
+
+basic_tls_options(Options) ->
```
Comment (kinyoklion) [inline test-service/src/ts_sdk_config_params.erl:258]: The erlang SDK does not have a basic TLS configuration depending on if the underlying OTP has a default configuration. So we set one here to ensure we are always using secure defaults for the contract tests.

**Comment by kinyoklion** [comment]
Tests are going to fail until the contract test harness is updated.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,12 +13,13 @@ on:
...
     strategy:
+      # We want to know the status of all OTP versions, not just if one of them fails.
+      fail-fast: false
       matrix:
         versions:
         - {os: 'ubuntu-20.04', otp: '21.x', rebar: '3.15.2'}
-        - {os: 'ubuntu-22.04', otp: '24.x', rebar: '3.18.0'}
         - {os: 'ubuntu-22.04', otp: '25.x', rebar: '3.18.0'}
-        - {os: 'ubuntu-22.04', otp: '26.x', rebar: '3.22.0'}
+        - {os: 'ubuntu-22.04', otp: '27.x', rebar: '3.23.0'}
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:22]: OTP 27 just came out, so add it to the test matrix and drop a couple other versions.

File: Makefile [hunk 5]
```diff
@@ -46,8 +47,8 @@ build-contract-tests:
 	@mkdir -p test-service/_checkouts
 	@rm -f $(CURDIR)/test-service/_checkouts/ldclient
 	@ln -sf $(CURDIR)/ $(CURDIR)/test-service/_checkouts/ldclient
-	@if [ "$(OTP_VER)" = "26.x" ]; then\
-		echo Dialyze for OTP 26;\
+	@if [ "$(ERL_VERSION)" -ge "26" ]; then\
```
Comment (kinyoklion) [inline Makefile:50]: The dialyzer settings changes in 26 and the new settings don't work earlier. This changes the 26 check to be a 26+ check.

**Review by cwaldren-ld — APPROVED** [review]

**Pull request merged**